src/engine/BMGame.php

```diff
@@ -59,46 +59,46 @@
  * @SuppressWarnings(PMD.UnusedPrivateField)
  */
 class BMGame {
-    // properties -- all accessible, but written as private to enable the use of
+    // properties -- all accessible, but written as protected to enable the use of
     //               getters and setters
-    private $gameId;                // game ID number in the database
-    private $playerIdArray;         // array of player IDs
-    private $nPlayers;              // number of players in the game
-    private $roundNumber;           // current round number
-    private $turnNumberInRound;     // current turn number in current round
-    private $activePlayerIdx;       // index of the active player in playerIdxArray
-    private $playerWithInitiativeIdx; // index of the player who won initiative
-    private $buttonArray;           // buttons for all players
-    private $activeDieArrayArray;   // active dice for all players
-    private $attack;                // array('attackerPlayerIdx',
+    protected $gameId;                // game ID number in the database
+    protected $playerIdArray;         // array of player IDs
+    protected $nPlayers;              // number of players in the game
+    protected $roundNumber;           // current round number
+    protected $turnNumberInRound;     // current turn number in current round
+    protected $activePlayerIdx;       // index of the active player in playerIdxArray
+    protected $playerWithInitiativeIdx; // index of the player who won initiative
+    protected $buttonArray;           // buttons for all players
+    protected $activeDieArrayArray;   // active dice for all players
+    protected $attack;                // array('attackerPlayerIdx',
```

Comment:
I have an outstanding note to myself to change all private variables and methods to protected unless there's a good reason to have them as private. I'm doing this as I touch various files, so it's likely that you'll see this sort of change happening as different files are changed.

As for the rationale, I know that we've discussed the pros and cons of using private vs protected as the default level. With all the stuff that I've been doing with inheritance of skills, it's proven to be somewhat frustrating to have private as the default level, especially with the way that PHP doesn't really enforce encapsulation correctly. Thus, I'm changing the default level to protected as I encounter instances.

Of course, since the change doesn't actually change behaviour in the absence of inheritance, it currently looks like the change is purely superficial, but I anticipate gains in the future.